network/endpoint_windows.go

```diff
@@ -76,6 +76,30 @@ func (nw *network) newEndpointImpl(
 	_ ipTablesClient,
 	epInfo *EndpointInfo,
 ) (*endpoint, error) {
+	if epInfo.NICType == cns.BackendNIC {
+		// step 1: disable VF
+		if err := DisableVFDevice(epInfo.PnPID, plc); err != nil { //nolint
+			return nil, errors.Wrap(err, "failed to disable VF device")
+		}
+
+		// step 2: dismount VF
+		if err := DisamountVFDevice(epInfo.PnPID, plc); err != nil { //nolint
+			return nil, errors.Wrap(err, "failed to dismount VF device")
+		}
+
+		// step 3: assign new PciID back to interfaceInfo
+		pnpDeviceID, err := GetPnPDeviceID(epInfo.PnPID, plc)
+		if err != nil {
```

Comment:
Can you point me to code where you return error to containerd when one of these commands fail? and also point me to code where you handle delete and roll back these powershell commands on failure.

Reply:
Also please add UT for these two scenarios

Reply:
if any command is failed to get executed, then it will return error;
We do not return this error to containerd; we just trigger deletion call